src/pages/DashboardPage/DashboardProfilePage/DashboardProfilePage.tsx

```diff
@@ -55,6 +63,40 @@ const DashboardProfilePage: React.FC<DashboardProfilePageProps> = (props) => {
         }
     }
 
+    const onDelete = async (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        setIsLoading(true)
+        if (auth.currentUser?.email) {
+           try {
+                const credential = EmailAuthProvider.credential(
+                    auth.currentUser.email, 
+                    password
+                )
```

Comment:
Must have fresh credentials for the user before they can be deleted from firebase